src/DataViews/PresetsDataView.cpp

```diff
@@ -196,9 +206,22 @@ void PresetsDataView::OnShowInExplorerRequested(const std::vector<int>& selectio
 }
 
 void PresetsDataView::OnDoubleClicked(int index) {
-  const PresetFile& preset = GetPreset(index);
+  PresetFile& preset = GetMutablePreset(index);
   if (app_->GetPresetLoadState(preset).state != PresetLoadState::kNotLoadable) {
-    app_->LoadPreset(preset);
+    app_->LoadPreset(preset).Then(main_thread_executor_.get(),
+                                  [&preset](ErrorMessageOr<void> result) {
+                                    if (!result.has_error()) preset.SetIsLoaded(true);
+                                  });
+  }
+}
+
+void PresetsDataView::OnLoadPresetCompleted(const std::string& preset_file_path) {
```

Comment:
`preset_file_path` should be `std::string_view`.

Reply:
Or `const std::filesystem::path&` ?!

Reply:
This function is only called when the loading was successful (if I understand the code). Then maybe I would put success in the name. OnLoadPresetSuccessful , OnLoadPresetCompletedSuccessfully

Reply:
Changed to use `const std::filesystem::path&` and also renamed the function as `OnLoadPresetSuccessful`. Thanks!